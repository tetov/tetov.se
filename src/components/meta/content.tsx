import React from "react";
import { Helmet } from "react-helmet";

type Prop = {
  title: string;
  excerpt?: string;
  lang?: string;
  type?: string;
  dateCreated?: string;
};

const MetaContent: React.FC<Prop> = ({
  title,
  excerpt,
  type,
  lang,
  dateCreated,
}) => (
  <Helmet>
    {lang && <html lang={lang} />}

    <title>{title}</title>
    <meta property="og:title" content={title} />

    {excerpt && <meta name="description" content={excerpt} />}
    {excerpt && <meta property="og:description" content={excerpt} />}

    {type && <meta property="og:type" content={type} />}

    {dateCreated && type === "article" && (
      <meta property="article:published_time" content={dateCreated} />
    )}
  </Helmet>
);

export default MetaContent;
