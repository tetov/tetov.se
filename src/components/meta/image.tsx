import { graphql, useStaticQuery } from "gatsby";
import React from "react";
import { Helmet } from "react-helmet";

type MetaImageProp = {
  src?: string;
  alt?: string;
};

type PureMetaImageProp = MetaImageProp & {
  siteURL: string;
};

export const PureMetaImage: React.FC<PureMetaImageProp> = ({
  src,
  alt,
  siteURL,
}) => (
  <Helmet>
    {src && <meta property="og:image" content={siteURL + src} />}
    {src && alt && <meta property="og:image:alt" content={alt} />}
  </Helmet>
);

const MetaImage: React.FC<MetaImageProp> = ({ src, alt }) => {
  const data = useStaticQuery<GatsbyTypes.SiteURLQuery>(graphql`
    query SiteURL {
      site {
        siteMetadata {
          siteURL
        }
      }
    }
  `);
  return (
    <PureMetaImage
      src={src}
      alt={alt}
      siteURL={data.site.siteMetadata.siteURL}
    />
  );
};

export default MetaImage;
