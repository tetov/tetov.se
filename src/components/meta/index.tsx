export { default as MetaContent } from "./content";
export { default as MetaImage } from "./image";
export { default as MetaSiteWide } from "./site-wide";
export type { MetaSiteWide as MetaProp } from "./site-wide";
