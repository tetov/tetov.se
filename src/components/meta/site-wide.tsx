import querySiteMetadata from "hooks/query-site-metadata";
import queryTwitterUname from "hooks/query-twitter-uname";
import logo from "images/logo.png";
import React from "react";
import { Helmet } from "react-helmet";

export type MetaSiteWide = { pathName: string };

const MetaSiteWide: React.FC<MetaSiteWide> = ({ pathName }) => {
  const { title, lang, siteURL, description } = querySiteMetadata();
  const twitterUsername = queryTwitterUname();
  return (
    <Helmet titleTemplate={`%s | ${title}`} defaultTitle={title}>
      <html lang={lang || "en"} prefix="og: http://ogp.me/ns#" />

      <link rel="micropub" href="https://micropub.tetov.xyz/micropub" />
      <link rel="authorization_endpoint" href="https://indieauth.com/auth" />
      <link rel="token_endpoint" href="https://tokens.indieauth.com/token" />

      <meta property="og:site_name" content={title} />

      <meta property="og:url" content={siteURL + pathName} />

      <meta property="og:type" content="website" />

      <meta property="og:title" content={title} />

      <meta name="description" content={description.trim()} />
      <meta property="og:description" content={description.trim()} />

      <meta property="og:image" content={siteURL + logo} />

      <meta name="twitter:card" content="summary" />
      <meta name="twitter:site" content={twitterUsername} />
      <meta name="twitter:creator" content={twitterUsername} />
    </Helmet>
  );
};

export default MetaSiteWide;
