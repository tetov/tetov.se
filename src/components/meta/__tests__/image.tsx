import React from "react";
import renderer from "react-test-renderer";
import { PureMetaImage as MetaImage } from "src/components/meta/image";

describe("MetaImage", () => {
  const test_src = "content/projs/adaptive_clay_formations/hero.jpg";
  const siteURL = "https://tetov.se/";
  const test_alt = "Alt text";
  it.each`
    src          | alt
    ${test_src}  | ${test_alt}
    ${undefined} | ${test_alt}
    ${test_src}  | ${undefined}
    ${undefined} | ${undefined}
  `("src: $src alt: $alt", ({ src, alt }) => {
    const tree = renderer
      .create(<MetaImage src={src} siteURL={siteURL} alt={alt} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
