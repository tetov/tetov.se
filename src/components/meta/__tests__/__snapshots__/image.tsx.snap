// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaImage src: content/projs/adaptive_clay_formations/hero.jpg alt: Alt text 1`] = `
<div
  className="mock-helmet"
>
  <meta
    content="https://tetov.se/content/projs/adaptive_clay_formations/hero.jpg"
    property="og:image"
  />
  <meta
    content="Alt text"
    property="og:image:alt"
  />
</div>
`;

exports[`MetaImage src: content/projs/adaptive_clay_formations/hero.jpg alt: undefined 1`] = `
<div
  className="mock-helmet"
>
  <meta
    content="https://tetov.se/content/projs/adaptive_clay_formations/hero.jpg"
    property="og:image"
  />
</div>
`;

exports[`MetaImage src: undefined alt: Alt text 1`] = `
<div
  className="mock-helmet"
/>
`;

exports[`MetaImage src: undefined alt: undefined 1`] = `
<div
  className="mock-helmet"
/>
`;
